working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1595,9 +1634,13 @@ To type check a pattern `p` being matched against a value of type `M`:
 
     [nonnull]: https://github.com/dart-lang/language/blob/master/accepted/2.12/nnbd/feature-specification.md#null-promotion
 
-*   **Constant**: Type check the pattern's value in context type `M`. *The
-    context type comes into play for things like type arguments and
-    int-to-double:*
+*   **Constant**: Nothing to do. When type-checking the constant expression,
+    use no context type.
+
+    *Constant patterns can only appear in refutable contexts, and we don't want
```

Comment:
This is the breaking change that I referred to (incorrectly) above.  Currently, this code works, and prints "equal" followed by "not equal":

```dart
T id<T>(T x) => x;

void test(int Function(int) f) {
  switch(f) {
    case id: print("equal");break;
    default: print("not equal");break;
  }
  switch(f as Object) {
    case id: print("equal");break;
    default: print("not equal");break;
  }
  print("Done");
}


void main() {
  test(id);
}
```

Reply:
OK, so I as I understand it, this is no longer a breaking change because we're sticking with using the matched value's type as the context type for inference on the constant? This should still print "equal" followed by "not equal".